### PR TITLE
Fix repository settings form template to work with redmine 4.x

### PR DESCRIPTION
**Pull request by mathprod**
Patch from comments in issue #742
(https://github.com/jbox-web/redmine_git_hosting/issues/742#issuecomment-600840141)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -42,7 +42,7 @@ group :development, :test do
   gem 'database_cleaner'
   gem 'factory_bot_rails', '< 5.0'
 
-  gem 'rubocop', require: false
+  gem 'rubocop', '~> 0.76.0', require: false
```
Comment (PowerKiKi) [inline Gemfile:45]: Why did you change this ?

```suggestion
  gem 'rubocop', require: false
```

File: app/views/repository_deployment_credentials/edit.html.slim [hunk 2]
```diff
@@ -1,5 +1,5 @@
 = labelled_form_for :repository_deployment_credential, @credential,
-                    url: repository_deployment_credential_path(@repository, @credential),
+                    url: repository_deployment_credential_path(@repository, @credential), authenticity_token: form_authenticity_token,
```
Comment (PowerKiKi) [inline app/views/repository_deployment_credentials/edit.html.slim:2]: should wrap line like so:
```suggestion
                    url: repository_deployment_credential_path(@repository, @credential),
                    authenticity_token: form_authenticity_token,
```

File: app/views/repository_deployment_credentials/new.html.slim [hunk 3]
```diff
@@ -1,5 +1,5 @@
 = labelled_form_for :repository_deployment_credential, @credential,
-                    url: repository_deployment_credentials_path(@repository),
+                    url: repository_deployment_credentials_path(@repository), authenticity_token: form_authenticity_token,
```
Comment (PowerKiKi) [inline app/views/repository_deployment_credentials/new.html.slim:2]:
```suggestion
                    url: repository_deployment_credentials_path(@repository),
                    authenticity_token: form_authenticity_token,
```

File: app/views/repository_git_config_keys/edit.html.slim [hunk 4]
```diff
@@ -1,5 +1,5 @@
 = labelled_form_for :repository_git_config_key, @git_config_key,
-                    url: repository_git_config_key_path(@repository, @git_config_key),
+                    url: repository_git_config_key_path(@repository, @git_config_key), authenticity_token: form_authenticity_token,
```
Comment (PowerKiKi) [inline app/views/repository_git_config_keys/edit.html.slim:2]:
```suggestion
                    url: repository_git_config_key_path(@repository, @git_config_key),
                    authenticity_token: form_authenticity_token,
```

File: app/views/repository_git_config_keys/new.html.slim [hunk 5]
```diff
@@ -1,5 +1,5 @@
 = labelled_form_for :repository_git_config_key, @git_config_key,
-                    url: repository_git_config_keys_path(@repository),
+                    url: repository_git_config_keys_path(@repository), authenticity_token: form_authenticity_token,
```
Comment (PowerKiKi) [inline app/views/repository_git_config_keys/new.html.slim:2]:
```suggestion
                    url: repository_git_config_keys_path(@repository),
                    authenticity_token: form_authenticity_token,
```

File: app/views/repository_git_extras/move.html.slim [hunk 6]
```diff
@@ -2,7 +2,7 @@ h2 = l(:label_move_repository, repo_name: @repository.gitolite_repository_name)
 
 .box
   = labelled_form_for :repository_mover, @move_repository_form,
-                      url: move_repository_git_extras_path(@repository),
+                      url: move_repository_git_extras_path(@repository), authenticity_token: form_authenticity_token,
```
Comment (PowerKiKi) [inline app/views/repository_git_extras/move.html.slim:5]:
```suggestion
                      url: move_repository_git_extras_path(@repository),
                      authenticity_token: form_authenticity_token,
```

File: app/views/repository_mirrors/edit.html.slim [hunk 7]
```diff
@@ -1,5 +1,5 @@
 = labelled_form_for :repository_mirror, @mirror,
-                    url: repository_mirror_path(@repository, @mirror),
+                    url: repository_mirror_path(@repository, @mirror), authenticity_token: form_authenticity_token,
```
Comment (PowerKiKi) [inline app/views/repository_mirrors/edit.html.slim:2]:
```suggestion
                    url: repository_mirror_path(@repository, @mirror),
                    authenticity_token: form_authenticity_token,
```

File: app/views/repository_mirrors/new.html.slim [hunk 8]
```diff
@@ -1,5 +1,5 @@
 = labelled_form_for :repository_mirror, @mirror,
-                    url: repository_mirrors_path(@repository),
+                    url: repository_mirrors_path(@repository), authenticity_token: form_authenticity_token,
```
Comment (PowerKiKi) [inline app/views/repository_mirrors/new.html.slim:2]:
```suggestion
                    url: repository_mirrors_path(@repository),
                    authenticity_token: form_authenticity_token,
```

File: app/views/repository_post_receive_urls/edit.html.slim [hunk 9]
```diff
@@ -1,5 +1,5 @@
 = labelled_form_for :repository_post_receive_url, @post_receive_url,
-                    url: repository_post_receive_url_path(@repository, @post_receive_url),
+                    url: repository_post_receive_url_path(@repository, @post_receive_url), authenticity_token: form_authenticity_token,
```
Comment (PowerKiKi) [inline app/views/repository_post_receive_urls/edit.html.slim:2]:
```suggestion
                    url: repository_post_receive_url_path(@repository, @post_receive_url),
                    authenticity_token: form_authenticity_token,
```

File: app/views/repository_post_receive_urls/new.html.slim [hunk 10]
```diff
@@ -1,5 +1,5 @@
 = labelled_form_for :repository_post_receive_url, @post_receive_url,
-                    url: repository_post_receive_urls_path(@repository),
+                    url: repository_post_receive_urls_path(@repository), authenticity_token: form_authenticity_token,
```
Comment (PowerKiKi) [inline app/views/repository_post_receive_urls/new.html.slim:2]:
```suggestion
                    url: repository_post_receive_urls_path(@repository),
                    authenticity_token: form_authenticity_token,
```

File: app/views/repository_protected_branches/edit.html.slim [hunk 11]
```diff
@@ -1,5 +1,5 @@
 = labelled_form_for :repository_protected_branche, @protected_branch,
-                    url: repository_protected_branch_path(@repository, @protected_branch),
+                    url: repository_protected_branch_path(@repository, @protected_branch), authenticity_token: form_authenticity_token,
```
Comment (PowerKiKi) [inline app/views/repository_protected_branches/edit.html.slim:2]:
```suggestion
                    url: repository_protected_branch_path(@repository, @protected_branch),
                    authenticity_token: form_authenticity_token,
```

File: app/views/repository_protected_branches/new.html.slim [hunk 12]
```diff
@@ -1,5 +1,5 @@
 = labelled_form_for :repository_protected_branche, @protected_branch,
-                    url: repository_protected_branches_path(@repository),
+                    url: repository_protected_branches_path(@repository), authenticity_token: form_authenticity_token,
```
Comment (PowerKiKi) [inline app/views/repository_protected_branches/new.html.slim:2]:
```suggestion
                    url: repository_protected_branches_path(@repository),
                    authenticity_token: form_authenticity_token,
```

File: Gemfile (hunk 1 above)
Comment (mathprod) [reply Gemfile]: Changed it just to stop bundle install complaint about the rubocop version (already defined in redmine 4.1x Gemfile).
But I agree it's more a dirty hack than a patch.

**Comment by mathprod** [comment]
Thank for merging this.
About the previous question: this patch was tested on the following setup (Debian10)

```
Environment:
  Redmine version                4.1.1.stable
  Ruby version                   2.5.5-p157 (2019-03-15) [arm-linux-gnueabihf]
  Rails version                  5.2.4.2
  Environment                    production
  Database adapter               Mysql2
  Mailer queue                   ActiveJob::QueueAdapters::AsyncAdapter
  Mailer delivery                smtp
SCM:
  Subversion                     1.10.4
  Git                            2.20.1
  Filesystem                     
  Xitolite                       2.20.1
Redmine plugins:
  additionals                    2.0.23
  redmine_bootstrap_kit          0.2.5
  redmine_code_review            1.0.0
  redmine_default_members        1.0-devel
  redmine_dmsf                   2.4.2 devel
  redmine_embedded               0.0.2
  redmine_git_hosting            4.0.1
  redmine_per_project_formatting 0.1.0
```
